### ISSUE TITLE
Docs: It's not clear how ignoring files works

### ISSUE THREAD

**Issue by steida** [eslint/eslint#19093]
### Docs page(s)

https://eslint.org/docs/latest/use/configure/ignore

### What documentation issue do you want to solve?

Take a look at this discussion. No one understands how ignoring works. Because it's so hard, Eslint, by default, lints wrong files, and instead of failing gracefully, it throws exceptions about some modules that can't be loaded.

https://github.com/eslint/eslint/discussions/18304

This is really bad DX.

### What do you think is the correct solution?

Fix docs; don't throw unknown errors.

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by mdjermanovic** [comment]
Hi @steida, thanks for the issue! 

> ### What do you think is the correct solution?
> Fix docs; don't throw unknown errors.

Can you please clarify the suggested solution?

* What should be fixed in the docs?
* What errors should ESLint not throw?

**Comment by steida** [comment]
Hi, 

As I wrote, because ignoring didn't work as expected, Eslint loaded babel.config.js and failed with this:

```
> eslint


Oops! Something went wrong! :(

ESLint: 9.14.0

Error: Error while loading rule '@typescript-eslint/await-thenable': You have used a rule which requires type information, but don't have parserOptions set to generate type information for this file. See https://typescript-eslint.io/getting-started/typed-linting for enabling linting with type information.
Parser: typescript-eslint/parser
Occurred while linting /Users/steida/dev/evolu-private/apps/native/babel.config.js
    at throwError (/Users/steida/dev/evolu-private/node_modules/@typescript-eslint/utils/dist/eslint-utils/getParserServices.js:38:11)
    at getParserServices (/Users/steida/dev/evolu-private/node_modules/@typescript-eslint/utils/dist/eslint-utils/getParserServices.js:27:9)
    at create (/Users/steida/dev/evolu-private/node_modules/@typescript-eslint/eslint-plugin/dist/rules/await-thenable.js:49:55)
    at Object.create (/Users/steida/dev/evolu-private/node_modules/@typescript-eslint/utils/dist/eslint-utils/RuleCreator.js:31:20)
    at createRuleListeners (/Users/steida/dev/evolu-private/node_modules/eslint/lib/linter/linter.js:943:21)
    at /Users/steida/dev/evolu-private/node_modules/eslint/lib/linter/linter.js:1068:84
    at Array.forEach (<anonymous>)
    at runRules (/Users/steida/dev/evolu-private/node_modules/eslint/lib/linter/linter.js:999:34)
    at #flatVerifyWithoutProcessors (/Users/steida/dev/evolu-private/node_modules/eslint/lib/linter/linter.js:1911:31)
    at Linter._verifyWithFlatConfigArrayAndWithoutProcessors (/Users/steida/dev/evolu-private/node_modules/eslint/lib/linter/linter.js:1992:49)

Oops! Something went wrong! :(

ESLint: 9.14.0

AbortError: The operation was aborted
    at checkAborted (node:internal/fs/promises:456:11)
    at Object.readFile (node:internal/fs/promises:1083:3)
    at retrier.retry.signal (/Users/steida/dev/evolu-private/node_modules/eslint/lib/eslint/eslint.js:803:47)
    at #call (/Users/steida/dev/evolu-private/node_modules/eslint/node_modules/@humanwhocodes/retry/dist/retrier.cjs:298:22)
    at /Users/steida/dev/evolu-private/node_modules/eslint/node_modules/@humanwhocodes/retry/dist/retrier.cjs:360:44
    at #processPending (/Users/steida/dev/evolu-private/node_modules/eslint/node_modules/@humanwhocodes/retry/dist/retrier.cjs:399:19)
    at /Users/steida/dev/evolu-private/node_modules/eslint/node_modules/@humanwhocodes/retry/dist/retrier.cjs:313:33
    at <anonymous>

Oops! Something went wrong! :(

ESLint: 9.14.0

AbortError: The operation was aborted
```

Then I tried to change something with ignored files (still no idea how it works)

```
evolu-private$ pn lint

> @evolu/monorepo@ lint /Users/steida/dev/evolu-private
> eslint


<--- Last few GCs --->

[51583:0x158008000]    31082 ms: Mark-Compact (reduce) 2040.8 (2086.1) -> 2039.7 (2086.4) MB, 547.38 / 0.00 ms  (average mu = 0.264, current mu = 0.056) allocation failure; scavenge might not succeed
[51583:0x158008000]    31775 ms: Mark-Compact (reduce) 2040.8 (2086.4) -> 2040.3 (2086.9) MB, 690.92 / 0.00 ms  (average mu = 0.160, current mu = 0.003) allocation failure; scavenge might not succeed


<--- JS stacktrace --->

FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
----- Native stack trace -----

 1: 0x102dd559c node::Abort() [/Users/steida/.nvm/versions/node/v20.11.1/bin/node]
 2: 0x102dd579c node::ModifyCodeGenerationFromStrings(v8::Local<v8::Context>, v8::Local<v8::Value>, bool) [/Users/steida/.nvm/versions/node/v20.11.1/bin/node]
 3: 0x102f5a95c v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, v8::OOMDetails const&) [/Users/steida/.nvm/versions/node/v20.11.1/bin/node]
 4: 0x10312f050 v8::internal::Heap::GarbageCollectionReasonToString(v8::internal::GarbageCollectionReason) [/Users/steida/.nvm/versions/node/v20.11.1/bin/node]
 5: 0x10312db2c v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/Users/steida/.nvm/versions/node/v20.11.1/bin/node]
 6: 0x103124344 v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/Users/steida/.nvm/versions/node/v20.11.1/bin/node]
 7: 0x103124ba4 v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/Users/steida/.nvm/versions/node/v20.11.1/bin/node]
 8: 0x103109c14 v8::internal::Factory::NewFillerObject(int, v8::internal::AllocationAlignment, v8::internal::AllocationType, v8::internal::AllocationOrigin) [/Users/steida/.nvm/versions/node/v20.11.1/bin/node]
 9: 0x1034f16dc v8::internal::Runtime_AllocateInYoungGeneration(int, unsigned long*, v8::internal::Isolate*) [/Users/steida/.nvm/versions/node/v20.11.1/bin/node]
10: 0x103850c44 Builtins_CEntry_Return1_ArgvOnStack_NoBuiltinExit [/Users/steida/.nvm/versions/node/v20.11.1/bin/node]
11: 0x10381fad4 Builtins_MapConstructor [/Users/steida/.nvm/versions/node/v20.11.1/bin/node]
12: 0x1037c58fc Builtins_JSBuiltinsConstructStub [/Users/steida/.nvm/versions/node/v20.11.1/bin/node]
13: 0x109fa68c4 
```

**Comment by steida** [comment]
And, for the love of God, is it really not possible to explain in a few simple sentences how ignores/files work and how it is merged between config items? Click on the link; it's really hard to grasp.

Anyway, thank you for your hard work. Appreciated. Software is hell.

**Comment by mdjermanovic** [comment]
I believe the source of confusion is how the `ignores` key works when it's used in a config object that has no other keys vs when it's used in a config object that has other keys, but I believe it's already explained in the docs:

https://github.com/eslint/eslint/blob/db0b844a66ee25483f9619d04346de1a2a0d79fa/docs/src/use/configure/ignore.md?plain=1#L20-L22

https://github.com/eslint/eslint/blob/db0b844a66ee25483f9619d04346de1a2a0d79fa/docs/src/use/configure/ignore.md?plain=1#L41-L43

https://github.com/eslint/eslint/blob/db0b844a66ee25483f9619d04346de1a2a0d79fa/docs/src/use/configure/configuration-files.md?plain=1#L64

https://github.com/eslint/eslint/blob/db0b844a66ee25483f9619d04346de1a2a0d79fa/docs/src/use/configure/configuration-files.md?plain=1#L209-L211

https://github.com/eslint/eslint/blob/db0b844a66ee25483f9619d04346de1a2a0d79fa/docs/src/use/configure/configuration-files.md?plain=1#L101-L103

https://github.com/eslint/eslint/blob/db0b844a66ee25483f9619d04346de1a2a0d79fa/docs/src/use/configure/configuration-files.md?plain=1#L149

**Comment by steida** [comment]
Sure, but what about the rest? It took me hours to make this simple, stupid config. Guess why. It's because docs do not have examples. Please put at least a few examples of _working_ configs there.

The current docs are confusing, overwhelmed by details, and without _working_ examples. Thank you.

```ts
import eslint from "@eslint/js";
import tseslint from "typescript-eslint";
import jsdoc from "eslint-plugin-jsdoc";

export default tseslint.config(
  {
    ignores: ["**/*.js", "**/*.mjs", "**/.next/", "**/.turbo/", "**/dist/"],
  },
  {
    files: ["apps/**/*.{ts,tsx}", "packages/**/*.{ts,tsx}"],
    extends: [
      eslint.configs.recommended,
      ...tseslint.configs.strictTypeChecked,
      ...tseslint.configs.stylisticTypeChecked,
      jsdoc.configs["flat/recommended"],
    ],
    plugins: {
      jsdoc,
    },
    languageOptions: {
      parserOptions: {
        projectService: true,
        tsconfigRootDir: import.meta.dirname,
      },
    },
    rules: {
```

**Comment by steida** [comment]
To be honest, I still don't know how ignores/files are resolved by each other. If I move extends and languageOptions to the upper object, eslint is slow as hell. How should I structure that array of items? No idea.

**Comment by mdjermanovic** [comment]
> If I move extends and languageOptions to the upper object, eslint is slow as hell or throws.

As noted in [Ignore Files](https://eslint.org/docs/latest/use/configure/ignore), in order to ignore files/directories, `ignores` key should be used without any other keys in the configuration object. When you move `languageOptions` to that object, you're no longer ignoring files.

@eslint/eslint-team any ideas on how we could further clarify this?

Maybe:

1. Bold "without any other keys" in the Ignore Files docs.
2. Add a "Files are not ignored" section to https://eslint.org/docs/latest/use/troubleshooting/

**Comment by fasttime** [comment]
> As noted in [Ignore Files](https://eslint.org/docs/latest/use/configure/ignore), in order to ignore files/directories, `ignores` key should be used without any other keys in the configuration object. When you move `languageOptions` to that object, you're no longer ignoring files.

I've seen a lot of confusion around ignoring files in ESLint, and it really boils down to the fact that the `ignores` key in a config has two different meanings, depending on the context.

> [@eslint/eslint-team](https://github.com/orgs/eslint/teams/eslint-team) any ideas on how we could further clarify this?
> 
> Maybe:
> 
>     1. Bold "without any other keys" in the Ignore Files docs.
> 
>     2. Add a "Files are not ignored" section to https://eslint.org/docs/latest/use/troubleshooting/

Makes sense to me, although a troubleshooting section directly on the [Ignore Files](https://eslint.org/docs/latest/use/configure/ignore) page (_Why are my files not ignored?_) would be more accessible.

**Comment by nzakas** [comment]
I think @mdjermanovic's two suggestions are fine, although I think part of the problem is that people aren't really reading the docs. They are just skimming over. At least with a troubleshooting page we can point people there whenever someone comes to us with this question.

@fasttime I would rather link off to a page in the troubleshooting section rather than creating a troubleshooting heading on the same page.

**Comment by TeaDrinkingProgrammer** [comment]
> I think [@mdjermanovic](https://github.com/mdjermanovic)'s two suggestions are fine, although I think part of the problem is that people aren't really reading the docs. They are just skimming over. At least with a troubleshooting page we can point people there whenever someone comes to us with this question.
> 
> [@fasttime](https://github.com/fasttime) I would rather link off to a page in the troubleshooting section rather than creating a troubleshooting heading on the same page.

I think that is a fair assessment, but to fair, the wording only indirectly implies that it needs to be seperate:
`In your eslint.config.js file, if an ignores key is used without any other keys in the configuration object, then the patterns act as global ignores.`

You need to read it, and then realise that if it is not put in a seperate configuration object, it doesn't work. It doesn't explicitly state that.

Something like this would be clearer IMO:


>You can ignore files using the `ignores` key. The pattern entered will act as a global ignore.
>The ignores key needs to be in a seperate configuration object in order to work, like in the example: 
> 
> ```js
> // eslint.config.js
> export default [
>     {
>         ignores: [".config/*"]
>     },
>     {
>        // other keys....
>     }
> ];
> ```
> Incorrect:
> ```js
> // eslint.config.js
> export default [
>     {
>         ignores: [".config/*"]
>         // other keys....
>     },
> ];
> ```

**Comment by photonstorm** [comment]
@TeaDrinkingProgrammer thank you for that example above! I had read the docs multiple times, but it made absolutely no logical sense why `ignores` would work one way when isolated in a config vs. another when used with other keys. That's like the ultimate JS ecosystem anti-pattern I can think of. Your example clarified it for me. I still don't understand why it was created like this, but at least I can actually ignore folders now. Your example should be in the docs.

**Comment by aderant-smcd** [comment]
> I've seen a lot of confusion around ignoring files in ESLint, and it really boils down to the fact that the `ignores` key in a config has two different meanings, depending on the context.

I think this is the core issue here - it's fine to expect people to read docs, but something like a file ignore setting brings with it a bunch of preexisting assumptions based on configs in other systems and of course configs in previous versions of eslint - and in this case those preexisting assumptions largely conflict with how it currently works. 

I think it would be sensible to deprecate the existing setting and split ignores up into two different new values: 'globalIgnores' and 'blockIgnores' (the latter probably needs a better name).
* 'globalIgnores' would *always* apply to all files, regardless of whether it was in a block by itself or alongside other items (and probably merge with other instances of the same value?).
* 'blockIgnores' would *always* apply to only rules in the current block.

This way it would be obvious to anyone coming from another system with similar settings that they need to read the docs to understand how ignores work.

I imagine it would be bad practice to include globalIgnores in shared plugins, but maybe there are use cases.

**Comment by nzakas** [comment]
Before going any further, it would be helpful to get a bit more information about what exactly is confusing.

To start, before this discussion, what did you expect this config to do?

```js
export default [
    {
        ignores: ["**/*.test.js"],
        rules: {
            semi: "off"
        }
    }
];
```

**Comment by TeaDrinkingProgrammer** [comment]
I would first have thought that that would be a global ignore, because I didn't know that igbore can also be used for a rule block.

**Comment by JoshuaKGoldberg** [comment]
I also think I'd have taken that to be a global ignore at first. A lot of other config systems are set up so that a top-level `exclude`/`ignores` does exactly that, and that config looks pretty similar at first glance. For example, https://vitest.dev/config/#configuring-vitest's docs include a top-level `exclude` right at the top of the page today.

In work on StackOverflow and in the tseslint Discord, I've found a good chunk of users to also think it's a global ignore. Here's a quick incomplete list of other questions that have stated similar confusion:

* https://github.com/eslint/eslint/discussions/18304#discussioncomment-9069706
* https://stackoverflow.com/questions/78866310/eslint-typescript-using-random-js-files-as-config-and-failing-them/78866363#78866363 (see comments)
* https://stackoverflow.com/questions/78873311/parsing-error-was-not-found-by-the-project-service-but-ive-ignored-these-file/78873727#comment139067128_78873727
* https://stackoverflow.com/questions/79108378/eslint-flat-configts-it-scans-unincluded-files
* typescript-eslint Discord > `#help` > https://discord.com/channels/1026804805894672454/1287440343775707168
* typescript-eslint Discord > `#help` > https://discord.com/channels/1026804805894672454/1283776448188121158

I'd been planning on filing a new issue suggesting either making a new property or splitting `ignores` into two: essentially https://github.com/eslint/eslint/issues/19093#issuecomment-2469247481. But since there's discussion here beyond just adding to docs I can hold off for now. 🙂

**Comment by TeaDrinkingProgrammer** [comment]
Thank you for the great @JoshuaKGoldberg. I didn't know that there was also a second discussion going on. Would it be a good idea to maks a merge request for the docs with my wording (or something similar) so it is slightly less confusing until there can be a consensus for a final solution?

**Comment by photonstorm** [comment]
This can be solved in the documentation. But it needs to be crystal clear, with a config sample that directly spells it out. Right now, issues like this are the way people are learning what to do. I checked the docs first, several times, but it never 'clicked' or made sense, because it was such a different pattern compared to what I'm used to elsewhere.

I even tried using your config inspector tool before looking here. I actually think you could surface the issue in there as a warning - it would be easy to check for a config that looks like it is trying to apply ignores but only on the rules level, and just flag it up to the user "Are you really trying to ignore globally?" - that would have saved me a trip here, too.

**Comment by nzakas** [comment]
> I actually think you could surface the issue in there as a warning - it would be easy to check for a config that looks like it is trying to apply ignores but only on the rules level, and just flag it up to the user "Are you really trying to ignore globally?" - that would have saved me a trip here, too.

That's a good suggestion. Please open an issue on the config inspector:
https://github.com/eslint/config-inspector/issues

**Comment by JoshuaKGoldberg** [comment]
> [@TeaDrinkingProgrammer](https://github.com/TeaDrinkingProgrammer) [@JoshuaKGoldberg](https://github.com/JoshuaKGoldberg) I understand you thought that `ignores` in my example was a global ignores, but what did you think `rules` would do in that context? That's what I'm trying to understand.

I'd have thought that `rules` would apply to every file still being included. As in, the config was essentially saying two things:
1. Ignore all `"**/*.test.js"` files
2. Enable the `semi` files on all (non-ignored) files

> I don't think this is a good analogy because we don't have a "top-level". We're exporting an array of objects that are merged, other tools export an object.

Are there commonly used tools other than ESLint that export a top-level array of settings, rather than an object? My perception is that because ESLint is so unique in this sense, folks are jumping to that incorrect analogy and getting confused when it doesn't match reality.

The closest equivalent I can think of is Knip in monorepos: https://knip.dev/features/monorepos-and-workspaces. But in that system it's object entries given an explicit key each.

**Comment by melroy89** [comment]
> Sure, but what about the rest? It took me hours to make this simple, stupid config. Guess why. It's because docs do not have examples. Please put at least a few examples of _working_ configs there.
> 
> The current docs are confusing, overwhelmed by details, and without _working_ examples. Thank you.
> 
> import eslint from "@eslint/js";
> import tseslint from "typescript-eslint";
> import jsdoc from "eslint-plugin-jsdoc";
> 
> export default tseslint.config(
>   {
>     ignores: ["**/*.js", "**/*.mjs", "**/.next/", "**/.turbo/", "**/dist/"],
>   },
>   {
>     files: ["apps/**/*.{ts,tsx}", "packages/**/*.{ts,tsx}"],
>     extends: [
>       eslint.configs.recommended,
>       ...tseslint.configs.strictTypeChecked,
>       ...tseslint.configs.stylisticTypeChecked,
>       jsdoc.configs["flat/recommended"],
>     ],
>     plugins: {
>       jsdoc,
>     },
>     languageOptions: {
>       parserOptions: {
>         projectService: true,
>         tsconfigRootDir: import.meta.dirname,
>       },
>     },
>     rules: {

It feels that even a rocket scientist is nothing in comparison to getting this new eslint flat config right.

**Comment by melroy89** [comment]
I also lost track @github-actions

**Comment by nzakas** [comment]
Not stale, we'll update the docs.

**Comment by gitriff** [comment]
> extends: [
>       eslint.configs.recommended,
>       ...tseslint.configs.strictTypeChecked,
>       ...tseslint.configs.stylisticTypeChecked,
>       jsdoc.configs["flat/recommended"],
>     ],

Thank you thank you, my savior.  Configuring eslint for the first time was terrible experience. Single `ingores` works somehow differently than `ingores` with other fields? You live and learn :D 

Also: `extends` exists?? All the offical documentation (https://typescript-eslint.io/getting-started/typed-linting/) just told to put it always as a first thing.

**Comment by thorn0** [comment]
It seems like people read the blog posts about the new config system but miss [the actual docs](https://eslint.org/docs/latest/use/configure/configuration-files#configuration-file), which are more helpful and straight to the point. There’s a link to the docs at the top, but it’s only in the second post. Maybe it could be made more prominent and added to all three.

Another idea could be to throw an error for objects with `ignores` if `Object.keys(obj).length > 1 && !obj.files`.

**Comment by gitriff** [comment]
FWIW I think throwing an error on invalid config is the best approach. I mean I looked at the official docs

```
export default [
    {
        files: ["src/**/*.js"],
        ignores: ["**/*.config.js"],
        rules: {
            semi: "error"
        }
    }
];
```

And I was like "yeah that is exactly what I have", but nothing got ignored. I never noticed: 

> Non-global ignores patterns can only match file names.

text in the docs. And even then for a new comer I don't think it is obvious what is non-global. Like the example above. Looks global to me... I don't know.

Also in the docs the first reference to `global` is 

> Non-global ignores patterns can only match file names.

at least it is still not clear to me what `global` means.

Any case if there is "ignore directories" in a place where the config doesn't do that, best thing is to fail with error message such as:

> Directories can only be ingored in global ignore block (here docs to ignore and here docs to what is global)

**Comment by thorn0** [comment]
@gitriff The first reference to "global" on that page is:

> If ignores is used without any other keys in the configuration object, then the patterns act as [global ignores](https://eslint.org/docs/latest/use/configure/configuration-files#globally-ignoring-files-with-ignores).

Looks like it'd make sense to link the word "global" in all the mentions of global and non-global ignores to https://eslint.org/docs/latest/use/configure/configuration-files#globally-ignoring-files-with-ignores

**Comment by nzakas** [comment]
We updated the docs in https://github.com/eslint/eslint/pull/19297.

This change will go live on the site some time today.

Let us know if you have any feedback on the changes.

**Comment by gitriff** [comment]
Thanks a lot. I think is is lot clearer now!

**Comment by nzakas** [comment]
I'm going to close this for the time being as we've made some updates. Docs can always be improved, so everyone should feel free to submit PRs if there's any way to make things clearer.

**Comment by anmpog** [comment]
I have to chime in, because I've found this thread in my pursuit of getting a very basic eslint configuration going using other people's plugins and I have to say - as someone who hasn't been using this tool for years, your documentation is really hard to get through. I wish there was a guide that showed how to build configurations for each of the existing approaches. Obviously there's still documentation around for the old `eslintrc` files/approach. But you're in the middle of migrating away from that and the documentation sort of just throws a TON of stuff at you and doesn't really paint a very clear picture of how to do things from scratch. For example, right now I have no idea if i should be using the `defineConfig` helper, or just defining an object a certain way. I have no idea what benefit is conferred by one approach or the other. I don't have a good idea of real-world use of the APIs you've provided. I will happily admit to not being the most savvy developer... but I feel like i'm the type of developer you should be aiming to "solve" for when you're working on documentation. So far, it's taken me 10+ hours of grappling with your documentation and trying various things and I STILL don't have a very clear idea of what I'm doing. The DX here is awful. Sure I'm sure plenty would disagree with me and they're welcome to... I'm no 10x engineer but I can usually figure out what the hell is going on with something after spending a while digging around in the docs. Not so with this tool and the current state of the documentation + the tool itself.

**Comment by nzakas** [comment]
@anmpog thanks for the feedback. We introduced `defineConfig()` in part to address the problems you're facing. The short version is that you should use `defineConfig()` because it makes your life easier. We're in the process of updating all of the documentation to mention it. Right now, this is the best place to start:
https://eslint.org/docs/latest/use/configure/configuration-files

This is for combining configs together:
https://eslint.org/docs/latest/use/configure/combine-configs

This is for ignoring files:
https://eslint.org/docs/latest/use/configure/ignore

If you have other thoughts or are still having trouble, please open a discussion so we can address them. (Closed issues don't get a lot of eyes on them, so opening a new discussion is preferable.)

**Comment by anmpog** [comment]
@nzakas Thanks for replying. I woke up today and kept trying to get it behaving in the way I wanted it to and I just don't have the energy to fight with it anymore. I'll have to check things out again in a few months once the documentation has some more polish and maybe some of the APIs are a little more clear to understand. Thank you for taking the time to reply, I've been digging around in threads and on the discord and notice you fielding a lot of complaints with grace and without getting defensive or snappy in return – just wanted to say i really respect that and thanks for working so hard on this! Ill make sure to send a small donation along.

**Comment by tacocode-dev** [comment]
@anmpog this works for me (`eslint.config.mjs`) using `globalIgnores`:

```ts
import { defineConfig, globalIgnores } from "eslint/config";
import { fixupConfigRules } from "@eslint/compat";
import globals from "globals";
import tsParser from "@typescript-eslint/parser";
import path from "node:path";
import { fileURLToPath } from "node:url";
import js from "@eslint/js";
import { FlatCompat } from "@eslint/eslintrc";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const compat = new FlatCompat({
  baseDirectory: __dirname,
  recommendedConfig: js.configs.recommended,
  allConfig: js.configs.all,
});

export default defineConfig([
  globalIgnores([
    ".vite/**/*",
  ]),
  {
    extends: fixupConfigRules(
      compat.extends(
        "eslint:recommended",
        "plugin:@typescript-eslint/eslint-recommended",
        "plugin:@typescript-eslint/recommended",
        "plugin:import/recommended",
        "plugin:import/electron",
        "plugin:import/typescript"
      )
    ),

    languageOptions: {
      globals: {
        ...globals.browser,
        ...globals.node,
      },

      parser: tsParser,
    },
  },
]);

```
